.github/workflows/data-fetch.yml

```diff
@@ -3,16 +3,27 @@ name: Data Fetch
 on:
   schedule:
   - cron: "0 7 * * *" # Every day at midnight PDT
+  pull_request:
+    branches:
+      - main
 
```

Comment:
I updated the script to run on PR since I didn't have access to the token locally.

We can revert changes in this file if not required. Or maybe push commit only if there are changes in the `tools/` directory.